prog/download_boot_image.rb

```diff
@@ -21,26 +49,32 @@ def blob_storage_client
   end
 
   label def start
-    vm_host.update(allocation_state: "draining")
-    hop_wait_draining
-  end
-
-  label def wait_draining
-    nap 15 unless vm_host.vms_dataset.where(boot_image: image_name).empty?
-
-    sshable.cmd("sudo rm -f /var/storage/images/#{image_name.shellescape}.raw")
+    fail "Image already exists on host" if vm_host.boot_images_dataset.where(name: image_name, version: version).count > 0
+    BootImage.create(
+      vm_host_id: vm_host.id,
+      name: image_name,
+      version: version,
+      active: false
+    ) { _1.id = UBID.generate(UBID::TYPE_ETC).to_uuid }
```

Comment:
If you override `self.ubid_type` method, you can use `create_with_id` directly.

https://github.com/ubicloud/ubicloud/blob/f5cea35e33df427dbe3402b19b31dba7039ade4d/model/storage_device.rb#L10-L12